mkdocs/docs/kubernetes-architecture.md

```diff
@@ -1,8 +1,105 @@
+
+---
 
 ## Control Plane
-- etcd
 
-## Worker Node
-A worker node is a VM or physical computer that provides a worker in a Kubernetes system
-- Kubelet: communicates with the control plane
+The **Control Plane** is the brain of the Kubernetes cluster. It manages the cluster's state and makes decisions about scheduling, scaling, and responding to events. The Control Plane has individual components running as pods on the node, each responsible for various tasks. 
+
+#### Key Components:
+- API Server - Acts as the front-end for the Kubernetes control plane. All interactions (kubectl, dashboard, etc.) go through this RESTful API. `kube-apiserver-<node-name>`
+
+- Controller Manager - Runs controllers that handle routine tasks like node health checks, replication, and endpoint management. `kube-controller-manager-<node-name>`
+
+- Scheduler - Assigns newly created pods to nodes based on resource availability and constraints. `kube-scheduler-<node-name>`
+
+- etcd - A distributed key-value store that holds all cluster data (state, configuration, etc.). `etcd-<node-name>`
+
+---
+
+## Worker Nodes
+
+Worker nodes are where your application containers actually run. Each node has the following components:
+
+#### Key Components:
+- Kubelet - An agent that runs on each node. It communicates with the API server and ensures containers are running as expected.
+
+- Container Runtime - Software responsible for running containers (e.g., Docker, containerd).
+
+- Kube-proxy - Handles network routing and load balancing for services within the cluster.
+
+---
+
+## How do these components work together?
+
+1. You submit a deployment via `kubectl`.
+2. The **API Server** receives the request.
+3. The **Scheduler** picks a suitable node.
+4. The **Controller Manager** ensures the desired state is maintained.
+5. The **Kubelet** on the chosen node pulls the container image and starts the pod.
+6. **Kube-proxy** ensures networking is set up so the pod can communicate.
+
+---
+
+## Minikube Architecture
+
+During this workshop we will be demonstrating Kubernetes using Minikube to create clusters and deploy resources. Minikube is a tool that lets you run a single-node Kubernetes cluster locally on your machine. It’s designed for developers and learners who want to experiment with Kubernetes without needing a full multi-node setup.
+
+
+### How standard Kubernetes architecture maps to Minikube:
+
+| Standard Kubernetes | Minikube  |
+|---------------------|---------------------|
+| Control Plane       | Runs inside the Minikube VM/container |
+| Worker Node         | Same VM/container acts as the worker node |
+| Kubelet             | Runs inside Minikube |
+| API Server          | Accessible via `kubectl` on your host |
+| etcd, Scheduler, Controller Manager | All run inside the Minikube VM |
+
+Minikube supports many standard Kubernetes features as well as third-party extensions in the form of [addons](https://minikube.sigs.k8s.io/docs/handbook/deploying/#addons).
+
```

Comment:
How much work would it be to include a diagram here?

Reply:
That would be great, or we can just use an image someone else has made and reference it.